libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -1231,6 +1280,50 @@ IotMqttError_t IotMqtt_Connect( const IotMqttNetworkInfo_t * pNetworkInfo,
         IotLogInfo( "New MQTT connection %p established.", pMqttConnection );
         /* Set the output parameter. */
         *pMqttConnection = pNewMqttConnection;
+
+        /* Fill in Time Interface function pointer. */
+        callbacks.getTime = getTimeMs;
+
+        /* Getting the free index from the MQTT connection to MQTT context mapping array
+         * to store the MQTT context for the newly established MQTT connection. */
+        contextIndex = _IotMqtt_getFreeIndexFromContextConnectionArray();
+
+        if( contextIndex < 0 )
+        {
+            IOT_SET_AND_GOTO_CLEANUP( IOT_MQTT_NO_MEMORY );
```

Comment:
Print Error logs indicating the updation of config helping to resolve this error.